src/TON/Cell.h

```diff
@@ -0,0 +1,185 @@
+// Copyright © 2017-2019 Trust Wallet.
+//
+// This file is part of Trust. The full Trust copyright notice, including
+// terms governing use, modification, and redistribution, is contained in the
+// file LICENSE at the root of the source code distribution tree.
+
+#pragma once
+
+#include "../Data.h"
+
+#include <memory>
+
+namespace TW::TON {
+
+/**
+ * Simplified implementation of TON Slice, Cell, etc. structures, for minimal represenatation and
+ * hash computation.
+ */
+
+/**
+ * Represents a Slice.
+ * Sample usage, bytes:
+ *   Slice s1 = Slice::createFromHex("123456")); // or createFromData(parse_hex("123456"));
+ *   std::cout << "slice, size: " << s1.size() << " data: " << s1.asBytesStr() << " hash: " <<
+ *     hex(s1.hash()) << std::endl; 
+ * Outputs: slice, size: 3 data: 123456 hash:
+ *   bf7cbe09d71a1bcc373ab9a764917f730a6ed951ffa1a7399b7abd8f8fd73cb4
+ */
+class Slice {
+public:
+    Slice();
+    Slice(const Slice& from);
+    virtual ~Slice() {}
+    /// Constructor methods.  May throw.
+    static Slice createFromData(const Data& data);
+    static Slice createFromHex(std::string const& dataStr);
+    static Slice createFromBits(const Data& data, size_t sizeBits);
+    static Slice createFromBitsStr(std::string const& dataStr, size_t sizeBits);
+    void appendBytes(const Data& data_in);
+    /// Append bytes, possible incomplete bytes at the end.  SizeBits should be equal to data.size() * 8, or less by at most 7.
+    void appendBits(const TW::Data& data_in, size_t sizeBits);
+    Data data() const { return _data; }
+    inline size_t size() const { return _data.size(); }
+    size_t sizeBits() const { return _sizeBits; }
+    std::string asBytesStr() const;
+    void serialize(TW::Data& data_inout);
+    Data hash() const;
+
+protected:
+    void appendBitsAligned(const TW::Data& data_in, size_t sizeBits);
+    void appendBitsNotAligned(const TW::Data& data_in, size_t sizeBits);
+
+private:
+    Data _data;
+    size_t _sizeBits;
+};
+
+/**
```

Comment:
let's move these comments to `README.md` in src/TON ?

Reply:
Good idea, done.